src/Silex/CallbackResolver.php

```diff
@@ -33,35 +33,49 @@ public function __construct(Container $app)
      */
     public function isValid($name)
     {
-        return is_string($name) && preg_match(static::SERVICE_PATTERN, $name);
+        return is_string($name) && (preg_match(static::SERVICE_PATTERN, $name) || isset($this->app[$name]));
     }
 
     /**
      * Returns a callable given its string representation.
      *
      * @param string $name
      *
-     * @return array A callable array
+     * @return callable A callable value
      *
      * @throws \InvalidArgumentException In case the method does not exist.
      */
     public function convertCallback($name)
     {
-        list($service, $method) = explode(':', $name, 2);
+        if (preg_match(static::SERVICE_PATTERN, $name)) {
+            list($service, $method) = explode(':', $name, 2);
 
-        if (!isset($this->app[$service])) {
-            throw new \InvalidArgumentException(sprintf('Service "%s" does not exist.', $service));
+            if (!isset($this->app[$service])) {
+                throw new \InvalidArgumentException(sprintf('Service "%s" does not exist.', $service));
+            }
+
+            return array($this->app[$service], $method);
+        }
+
+        if (!isset($this->app[$name])) {
```

Comment:
I reword my previous comment: since you call the service line 57 which would throw a pimple exception, why not change the test line 53 to `if (!is_callable(array($this->app[$service], $method)))`?